saleor/product/search.py

```diff
@@ -44,46 +38,40 @@ def update_products_search_vector(products: "QuerySet"):
 
         prefetch_related_objects(products_batch, *PRODUCT_FIELDS_TO_PREFETCH)
         for product in products_batch:
-            product.search_vector = prepare_product_search_vector_value(
-                product, already_prefetched=True
+            product.search_vector = FlatConcat(
+                *prepare_product_search_vector_value(product, already_prefetched=True)
             )
 
         Product.objects.bulk_update(products_batch, ["search_vector", "updated_at"])
 
 
 def update_product_search_vector(product: "Product"):
-    product.search_vector = prepare_product_search_vector_value(product)
+    product.search_vector = FlatConcat(*prepare_product_search_vector_value(product))
     product.save(update_fields=["search_vector", "updated_at"])
 
 
 def prepare_product_search_vector_value(
     product: "Product", *, already_prefetched=False
-) -> NoValidationSearchVector:
+) -> List[NoValidationSearchVector]:
     if not already_prefetched:
         prefetch_related_objects([product], *PRODUCT_FIELDS_TO_PREFETCH)
-    search_vector = NoValidationSearchVector(
-        Value(product.name), config="simple", weight="A"
-    ) + NoValidationSearchVector(
-        Value(product.description_plaintext), config="simple", weight="C"
-    )
-    attributes_vector = generate_attributes_search_vector_value(
-        product.attributes.all()
-    )
-    if attributes_vector:
-        search_vector += attributes_vector
-    variants_vector = generate_variants_search_vector_value(product)
-    if variants_vector:
-        search_vector += variants_vector
-
-    return search_vector
+    search_vectors = [
+        NoValidationSearchVector(Value(product.name), config="simple", weight="A"),
+        NoValidationSearchVector(
+            Value(product.description_plaintext), config="simple", weight="C"
+        ),
+        *generate_attributes_search_vector_value(product.attributes.all()),
+        *generate_variants_search_vector_value(product),
+    ]
+    return search_vectors
 
 
 def generate_variants_search_vector_value(
     product: "Product",
-) -> Optional[NoValidationSearchVector]:
+) -> List[NoValidationSearchVector]:
```

Comment:
If there is possibility that a list returned will be empty shouldn't we use `List[Optional[NoValidationSearchVector]]` ? The same goes to all other methods that generates list of `NoValidationSearchVector`

Reply:
There is no null values inside the list, the typing is valid for empty lists